### DISCUSSION TITLE
Feasibility: Switching from Single-Column to Two-Column on Same Page? (Data Sheets)

### DISCUSSION THREAD

**Discussion by bai-yi-bai** [Mogztter/asciidoctor-web-pdf#694] [Q&A]
Hello,

I'm investigating whether it is possible to create a template which allows changing the number of columns dynamically within a single page. For example, from two-columns, to one-column, and back to two-columns. This is especially important for data sheets which are used in a wide variety of industries.

I've been rolling out AsciiDoc and Antora in my organization, but I've been getting push back because stakeholders want to maintain our existing data sheet presentation which requires changing the number of columns on a single page. These data sheets are laid out by hand using commercial tools. Typically, data sheets which might start with two-columns at the top of the cover page, then change to having a figure taking up the full width of the page, and then have a full page of text or a table underneath the figure at the bottom of the page. It's not unusual to have data sheets change column layout 3-5 times on a page. Yes, I agree it's not ideal and sometimes makes reading them difficult, but this is the traditional way things have been done for decades. Page 1 and 2 of this example data sheet illustrates the layout problem: [https://www.ti.com/lit/ds/symlink/sn74hc595.pdf](Texas Instruments SNx4HC595 8-Bit Shift Register With 3-State Output Register)

asciidoctor-web-pdf's resume example gave me some hope that this might be possible. I'd really like to be able to use AsciiDoc's include:: function with .csv files to accelerate the creation of data sheets since technical writers spend a lot of time mindlessly converting data from engineers/scientists. This process introduces a lot of errors. In addition, commercial tools make things difficult by embedding styling and text into a single document. Decoupling this would allow updating templates and branding much easier.

Before I attempt to create my own template (and css), I wanted to ask some questions on the feasibility of this idea.

1. It is possible to change column layout mid-page with asciidoctor-web-pdf? 
2. If it is not possible to control the column layout dynamically, perhaps I could define a few dozen mixed-column page layouts? This would allow designers/technical writers a few choices when designing pages for data sheets.

If either option is possible, I might be able to create an example data sheet for inclusion in this project; but this might take some time.

**Comment by ggrossetie** [comment]
> Page 1 and 2 of this example data sheet illustrates the layout problem

You can use a section with a two-column layout and another section with the illustration (full width). 
Depending on what you need to achieve you can also take a look at https://developer.mozilla.org/en-US/docs/Web/CSS/column-span

The [CSS multi-column layout](https://developer.mozilla.org/en-US/docs/Web/CSS/CSS_multicol_layout) does a decent job when you want to flow the content across multiple columns. If you need a more precise positioning or layout you might need to use CSS grid or Flexbox (in addition to using AsciiDoc roles to identify blocks or using a custom template).

Keep in mind that you can also run JavaScript during the rendering to update the layout (that's basically what Paged.js is doing).

In summary, you have many options but I cannot really tell you which one would be better.

> If it is not possible to control the column layout dynamically, perhaps I could define a few dozen mixed-column page layouts? 

I've implemented this strategy for a client in order to produce PDF presentation. It worked well. On each slide, writers were able to define which layout they wanted to use.
